tasks/release.py

```diff
@@ -152,3 +157,43 @@ def release_integration(ctx, target, dry_run=False):
             ctx.run(cmd, warn=True)
 
     print("Done.")
+
+
+@task(help={
+    'dest_path': "The path to the destination file, using stdout if empty"
+})
+def compile_requirements(ctx, dest_path=None):
+    """
+    Write the `agent_requirements.txt` file at the root of the repo listing
+    all the agent based integrations pinned at the version they currently
+    have in HEAD.
+    """
+    all_platforms = sorted(['linux', 'mac_os', 'windows'])
+
+    entries = []
+    for i in AGENT_BASED_INTEGRATIONS:
+        if i in AGENT_V5_ONLY:
+            print("Integration {} is only shipped with version 5 of the Agent, skip...")
+            continue
+        version = get_version_string(i)
+        m = load_manifest(i)
+        platforms = sorted(m.get('supported_os', []))
+        # all platforms
+        if platforms == all_platforms:
+            entries.append('{}=={}'.format(i, version))
+        # one specific platform
+        elif len(platforms) == 1:
+            entries.append("{}=={}; sys_platform == '{}'".format(i, version, platforms[0]))
+        # assuming linux+mac here for brevity
+        elif 'windows' not in platforms:
+            entries.append("{}=={}; sys_platform != 'windows'".format(i, version))
+        else:
+            print("Can't parse the 'supported_os' list for the check {}: {}".format(i, platforms))
+
+    output = '\n'.join(entries)
```

Comment:
Just in case we have `AGENT_BASED_INTEGRATIONS` out of order, let's do `sorted(entries)`.